frontend/src/components/AgentControlBar.tsx

```diff
@@ -72,13 +76,24 @@ function ActionButton({
 
 ActionButton.defaultProps = {
   large: false,
+  isLoading: false,
 };
 
 function AgentControlBar() {
+  const { initialized } = useSelector((state: RootState) => state.task);
   const { curTaskState } = useSelector((state: RootState) => state.agent);
   const [desiredState, setDesiredState] = React.useState(AgentTaskState.INIT);
   const [isLoading, setIsLoading] = React.useState(false);
 
+  const handleReset = () => {
+    if (!initialized) return;
+
+    initializeAgent(getSettings());
+    // act = AgentTaskAction.STOP;
+    clearMsgs().then().catch();
+    store.dispatch(clearMessages());
```

Comment:
Avoid importing the store directly into components. Always use hooks to interact with the store (`useDispatch` in this case)

Reply:
https://redux.js.org/faq/store-setup#can-or-should-i-create-multiple-stores-can-i-import-my-store-directly-and-use-it-in-components-myself